packages/datx/src/helpers/collection.ts

```diff
@@ -45,7 +45,11 @@ export function upsertModel(
     const parsedData = {};
 
     keys.forEach((key: string) => {
-      parsedData[key] = modelMapParse(TypeModel, data, key);
+      const isBackRefKey = Boolean(fields[key]?.referenceDef?.property);
+      const result = modelMapParse(TypeModel, data, key);
+      if (!(isBackRefKey && result === undefined && !(key in data))) {
+        parsedData[key] = result;
+      }
     });
```

Comment:
If the key is an indirect property and the calculated value is undefined and is not defined on the original data, the update of this value is ignored

if  `e` is an indirect property, and  `data` is ({a,b,c}) 
then current update chain is 
`({a , b, c}) ===>  ({a,b,c,d,e:undefined}) ===> ({a,b,c,d, e:undefined})`
This is incorrect, because I didn't specify that I wanted to update the 'e' property, but I actually updated the node with `undefined`( calculated by `modelMapParse` ) . The update mode is not consistent with `Object.assign`

the correct update chain maybe:
1.  `({a , b, c}) ===>  ({a,b,c,d,e:undefined}) ===> ({a,b,c,d})`
2.  `({a , b, c, e : undefined }) ===>  ({a,b,c,d,e:undefined}) ===> ({a,b,c,d, e : undefined })`
3.  `({a , b, c}) ===>  ({a,b,c,d,e: xxx}) ===> ({a,b,c,d, e:xxx})`

just link `Object.assign`